packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts

```diff
@@ -371,10 +277,19 @@ export class ObjectMetadataService extends TypeOrmQueryService<ObjectMetadataEnt
     const workspaceDataSource =
       await this.typeORMService.connectToDataSource(dataSourceMetadata);
 
-    await this.createObjectRelationsMetadataAndMigrations(
-      objectMetadataInput,
-      createdObjectMetadata,
-    );
+    if (isCustom) {
+      await this.createObjectRelationsMetadataAndMigrations(
+        objectMetadataInput,
+        createdObjectMetadata,
+      );
+    } else {
+      await this.remoteTableRelationsService.createAll(
+        objectMetadataInput.workspaceId,
+        createdObjectMetadata,
+        objectMetadataInput.primaryKeyFieldMetadataSettings,
+        objectMetadataInput.primaryKeyColumnType,
+      );
+    }
 
     await this.workspaceMigrationRunnerService.executeMigrationFromPendingMigrations(
```

Comment:
Not specifically related to your PR but im curious on how we handle migration errors. I feel sometimes the migration execution is performed in a try/catch, sometimes it is not, etc. would we have a problem if the migration failed here, after the fields rows have been added to fieldMetadata but not created in the tables for instance?

Reply:
We would, yes. Everything should be wrapped into a global transaction here, since it calls several service. We would probably need a better system to manage this